lib/semantics/resolve-names.cc

```diff
@@ -2152,6 +2148,43 @@ void ResolveNamesVisitor::Post(const parser::CallStmt &) {
   expectedProcFlag_ = std::nullopt;
 }
 
+bool ResolveNamesVisitor::Pre(const parser::ImportStmt &x) {
+  auto kind{MapImportKind(x.kind)};
+  auto &scope{currScope()};
+  // Check C896 and C899
```

Comment:
Could you summarize C896/C899 briefly here, please?

Reply:
OK. The specific constraints are obvious from the error messages, but I can add a one-line summary.